### PR TITLE
refactor SDK and add types to URL

### PR DESCRIPTION
**Pull request by karthik2804**
Signed-off-by: karthik Ganeshram <karthik.ganeshram@fermyon.com>

### CODE REVIEW

File: crates/spin-js-engine/src/js_sdk/modules/statusTextList.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
+/** @internal */
+export const statusTextList: { [key: string]: string } = {
```
Comment (dicej) [inline crates/spin-js-engine/src/js_sdk/modules/statusTextList.ts:2]: Should this be: `[key: number]` instead?

**Review by dicej — COMMENTED** [review]
Looks great -- just one question above.

File: crates/spin-js-engine/src/js_sdk/modules/statusTextList.ts (hunk 1 above)
Comment (karthik2804) [reply crates/spin-js-engine/src/js_sdk/modules/statusTextList.ts:2]: I believe that all the keys in javascript are strings. To make it clearer, I think I can wrap all the keys with `"` if that would make sense.

Comment (dicej) [reply crates/spin-js-engine/src/js_sdk/modules/statusTextList.ts:2]: No worries -- I guess if TypeScript knows what you mean, it's okay.

**Review by dicej — APPROVED** [review]

**Pull request merged**